### PR TITLE
Added code to add tags to traces, addTag and removeTag

### PR DESCRIPTION
**Pull request by gertjanal**

### CODE REVIEW

File: src/modules/traceContext.js [hunk 1]
```diff
@@ -35,6 +36,38 @@ class TraceContext {
...
+     * Add a tag to this trace.
+     *
+     * @param {string} tag Text to be added as tag, usually a single word
+     * @returns The promise of the response
+     */
+    addTag = (tag) => {
+        return this.sessionManager.gatekeeper(`/projects/${this.collectionId}/traces/${this.traceUid.traceUid}/tags/${window.encodeURIComponent(tag)}`, {
+            method: 'PUT',
+            headers: {
+                'Hansken-Project-Refresh': 'refresh',
```
Comment (bartbroere) [inline src/modules/traceContext.js]: What is the default for this on the server side, and should we make this overridable here, as an optional argument to this method?

**Review by bartbroere — APPROVED** [review]

File: examples/tags.html [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<script type="module">
+    import {HanskenClient} from './src/hansken-js.js';
+    const hansken = new HanskenClient('/gatekeeper', '/keystore');
+
+    const projectId = 'de554b81-e4a3-4759-96ec-0abf942be72c';
+    const traceUid = '1a85b2d2-972a-4131-86f7-f55b61b19158:0-0-10-3';
+
+    const traceContext = hansken.project(projectId).trace(traceUid);
+    traceContext.addTag('hello', false);
```
Comment (nfi-jd) [inline examples/tags.html]:
```suggestion
    traceContext.addTag('hello', {refresh: false});
```

File: src/modules/traceContext.js [hunk 3]
```diff
@@ -35,6 +36,44 @@ class TraceContext {
...
+    /**
+     * Add a tag to this trace.
+     *
+     * @param {string} tag Text to be added as tag, usually a single word
+     * @param {boolean} refresh Default set to `true` to refresh the project index after the tag is added.
+     *     This makes sure the tag is directly searchable, but is also slower for larger add operations.
+     *     In case of large amounts of add operations, it is better to set refresh to `false` and let Hansken decide when the index should be refreshed or call refresh on the last added tag.
+     * @returns The promise of the response
+     */
+    addTag = (tag, refresh = true) => {
```
Comment (nfi-jd) [inline src/modules/traceContext.js]:
```suggestion
    addTag = (tag, {refresh = true}) => {
```

File: src/modules/traceContext.js [hunk 4]
```diff
@@ -35,6 +36,44 @@ class TraceContext {
...
+    /**
+     * Remove a tag from this trace.
+     *
+     * @param {string} tag Text to be removed as tag, usually a single word
+     * @param {boolean} refresh Default set to `true` to refresh the project index after the tag is removed.
+     *     This makes sure the tag is directly removed from the search results, but is also slower for larger remove operations.
+     *     In case of large amounts of remove operations, it is better to set refresh to `false` and let Hansken decide when the index should be refreshed or call refresh on the last removed tag.
+     * @returns The promise of the response
+     */
+    removeTag = (tag, refresh = true) => {
```
Comment (nfi-jd) [inline src/modules/traceContext.js]:
```suggestion
    removeTag = (tag, {refresh = true}) => {
```

**Review by nfi-jd — APPROVED** [review]
I'd suggest using an options object for the refresh parameter. That way it's clear from the caller side what `false` means.

File: src/modules/traceContext.js (hunk 1 above)
Comment (gertjanal) [reply src/modules/traceContext.js]: The server default is `no-refresh`, but in frontends it is `refresh` by default. In python (batches) the default is no-refresh I think...

Comment (gertjanal) [reply src/modules/traceContext.js]: I updated the code so the refresh is configurable btw

**Pull request merged**